appinventor/components/src/com/google/appinventor/components/runtime/WebViewer.java

```diff
@@ -205,9 +206,11 @@ public View getView() {
   // this do nothing will override the default WebView behavior.  Returning
   // false means to let the WebView handle the Url.  In other words, returning
   // true will not follow the link, and returning false will follow the link.
+  // 
   private class WebViewerClient extends WebViewClient {
     @Override
     public boolean shouldOverrideUrlLoading(WebView view, String url) {
+      BeforePageLoad(url);
```

Comment:
We need to check the Android documentation that these methods will be called from the main / UI thread.